pip/wheel.py

```diff
@@ -602,8 +678,38 @@ def build(self):
         with indent_log():
             build_success, build_failure = [], []
             for req in buildset:
-                if self._build_one(req):
+                if autobuilding:
+                    output_dir = _cache_for_filename(
+                        self._cache_root, req.link.filename)
+                    ensure_dir(output_dir)
+                else:
+                    output_dir = self._wheel_dir
+                wheel_file = self._build_one(req, output_dir)
+                if wheel_file:
                     build_success.append(req)
+                    if autobuilding:
+                        # XXX: This is mildly duplicative with prepare_files,
+                        # but not close enough to pull out to a single common
+                        # method.
+                        # The code below assumes temporary source dirs -
+                        # prevent it doing bad things.
+                        if req.source_dir and not os.path.exists(os.path.join(
+                                req.source_dir, PIP_DELETE_MARKER_FILENAME)):
+                            raise Exception("bad source dir - missing marker")
```

Comment:
We don't want a bare ``Exception`` here, either an ``assert`` if we don't expect it to happen and it's just a guard or a more descriptive exception type.